### PR TITLE
RavenDB-10776 In Queries should not yield duplicate results

### PR DESCRIPTION
**Pull request by talweiss1982**

### CODE REVIEW

File: src/Raven.Server/Documents/Queries/CollectionQueryEnumerable.cs [hunk 1]
```diff
@@ -409,7 +409,7 @@ public override void VisitIn(QueryExpression fieldName, List<QueryExpression> va
                         {
                             if (item is ValueExpression iv)
                             {
-                                foreach (var id in QueryBuilder.GetValues(_query, _metadata, parameters, iv))
+                                foreach (var id in QueryBuilder.GetValues(_query, _metadata, parameters, iv, distinct:true))
                                 {
                                     AddId(id.Value?.ToString());
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Queries/CollectionQueryEnumerable.cs:414]: @talweiss1982 @ayende Wouldn't it be better to change the `List<Slice> Ids` to `HashSet<Slice>(SliceComparer.Instance)`?

Comment (ayende) [reply src/Raven.Server/Documents/Queries/CollectionQueryEnumerable.cs:414]: Yes, probably

Comment (talweiss1982) [reply src/Raven.Server/Documents/Queries/CollectionQueryEnumerable.cs:414]: not sure if the logic of the distinct is right for the general case, using an hashset mean we will do distinct on all values coming from different parameters.

Comment (ppekrol) [reply src/Raven.Server/Documents/Queries/CollectionQueryEnumerable.cs:414]: I think in all cases when we are doing collection query then we want to return distinct results?

Comment (ayende) [reply src/Raven.Server/Documents/Queries/CollectionQueryEnumerable.cs:414]: agreed

**Pull request merged**